diff --git a/src/Disturbance.hh b/src/Disturbance.hh
index c5e0106..803dba7 100644
--- a/src/Disturbance.hh
+++ b/src/Disturbance.hh
@@ -110,15 +110,12 @@ public:
     	std::vector<abs_type> ub(ss_dim);  /* upper-right corner */
     	std::vector<abs_type> no(ss_dim);  /* number of cells per dim */
     	std::vector<abs_type> cc(ss_dim);  /* coordinate of current cell in the (x,r) */
-        state_type lower_left;
-    	state_type upper_right;
-    	state_type m_eta;
-    	state_type m_z;
+        auto lower_left = states_alphabet.get_lower_left();
+        auto upper_right = states_alphabet.get_upper_right();
+        auto m_eta = states_alphabet.get_eta();
+        auto m_z = m_eta;
         for(int i=0; i<ss_dim; i++) {
-      		m_eta[i]=states_alphabet.get_eta()[i];
-      		m_z[i]=states_alphabet.get_eta()[i]/1e10;
-      		lower_left[i]=states_alphabet.get_lower_left()[i];
-      		upper_right[i]=states_alphabet.get_upper_right()[i];
+            m_z[i]=m_z[i]/1e10;
       	}
         abs_type nintersection=1;
         for(int k=0; k<ss_dim; k++) {
@@ -162,4 +159,4 @@ public:
 };//class closed
 }//namespace closed
 
-#endif
\ No newline at end of file
+#endif
